File: SimpleHabit/app/src/main/java/com/me/simplehabit/data/models/TopicModel.java
package com.me.simplehabit.data.models;

import com.me.simplehabit.data.vos.TopicVO;

import java.util.List;

public interface TopicModel {

    List<TopicVO> getTopics(boolean isForce,TopicModelDelegate delegate);

    interface TopicModelDelegate{
        void onTopicFetchFromNetwork(List<TopicVO> topicList);
        void onErrorTopicFetchFromNetwork(String message);
    }
}

File: SimpleHabit/app/src/main/java/com/me/simplehabit/data/models/TopicModel.kt
package com.me.simplehabit.data.models

import androidx.lifecycle.LiveData
import com.me.simplehabit.data.vos.TopicVO

interface TopicModel {

    fun getTopics(isForce: Boolean, networkFailure: (String) -> Unit): LiveData<List<TopicVO>>
}
